src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -134,3 +151,15 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
         }
     }
 }
+
+async function validateDatabaseName(name: string, client: PostgreSQLManagementClient, serverName: string, resourceGroup: string): Promise<string | undefined | null> {
+    const currentDatabaseList: DatabaseListResult = await client.databases.listByServer(resourceGroup, serverName);
```

Comment:
Similar comment to other PR. You want to do all sync calls first before you ever attempt an async call like this which could take longer. This method is executed every time the user types a character.

In fact, it would be best if we could find a way not to re-fetch the list of databases at all in this validate function. For example, if they've already expanded the tree, we should already know which databases there are.

Reply:
So, maybe caching the names of the current databases when we call loadMoreChildren...? Or creating a property for the list of databases, and then use it for loadMoreChildren and createChild/validate function?

Reply:
There's already a caching mechanism for tree items. Just use `getCachedChildren`. You can create the promise before `ext.ui.showInputBox({`:
```
const getChildrenTask: Promise<AzExtTreeItem[]> = this.getCachedChildren(context);
```

pass the task in to `validateDatabaseName`, but only `await` it after you've validated the other quick stuff like length